### PR TITLE
Make the writer even more unsafe.

### PR DESCRIPTION
**Pull request by jrmuizel**
Instead of changing the length every write we just adjust the pointer.
This avoids https://github.com/rust-lang/rust/issues/45068.
However we now need to ensure that we set the length when we are done.

---
This change is [Reviewable](https://reviewable.io/reviews/servo/webrender/1830)

### CODE REVIEW

File: webrender_api/src/display_list.rs [hunk 1]
```diff
@@ -523,7 +521,11 @@ fn serialize_fast<T: Serialize>(vec: &mut Vec<u8>, e: &T) {
     bincode::serialize_into(&mut size,e , bincode::Infinite).unwrap();
     vec.reserve(size.0);
 
-    bincode::serialize_into(&mut UnsafeVecWriter(vec), e, bincode::Infinite).unwrap();
+    let old_len = vec.len();
+    let ptr = unsafe { vec.as_mut_ptr().offset(vec.len() as isize) };
```
Comment (Gankro) [inline webrender_api/src/display_list.rs]: r=me with this changed to old_len

**Comment by jrmuizel** [comment]
I fixed up old_len and added a debug_assert to make sure things are sane.

**Comment by jrmuizel** [comment]
With this patch only 1.2% of WebRender display list building is spent in serialization.

**Comment by Gankro** [comment]
Nice!

**Comment by pcwalton** [comment]
What was the fraction of time spent in serialization before?

**Comment by jrmuizel** [comment]
I have too few samples to get a good comparison of the unsafe serializers. We have some patches coming that get rid of the cpu time spent elsewhere so that should raise the percentage of time spent in serialization. The safe serializer accounted for roughly 5.1% of the display list construction time.

**Comment by Gankro** [comment]
r=me, don't have review rights yet

**Pull request merged**